### PR TITLE
Allow configuration of JVM heap size

### LINKED ISSUE #94: mlockall is off by default..
I've verified that it's off by default https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-configuration-memory.html#mlockall

### LINKED ISSUE #167: [elastic6] Azure-repository settings should be set using keystore
After deploying the elastic6 template, it is currently impossible to restore backups. This is caused by a known issue(https://github.com/elastic/elasticsearch/issues/28299#issuecomment-358703209), but the proper solution is to save the azure.client.default.account and azure.client.default.key in the keystore isntead of elasticsearch.yml.

### PR DESCRIPTION
**Pull request by russcam**
This PR 

- allows the configuration of the JVM heap size with the introduction of a `esHeapSize` input parameter to the ARM template.
- stores Azure Cloud plugin credentials in the keystore for 6.x
- adds support for 6.2.2 and 5.6.8
- adds support for Elasticsearch 2.4.6 and Kibana 4.6.6
- enables bootstrap memory lock. Swap is disabled by default on Ubuntu Azure VMs, but no harm in setting memory lock in the template.

Closes #167 
Closes #94

### CODE REVIEW

**Comment by russcam** [comment]
links need to be updated to `master` before merging. Would rebase and merge over squashing

File: README.md [hunk 1]
```diff
@@ -121,6 +121,11 @@ Check out our [examples repository](https://github.com/elastic/azure-marketplace
     <td>Additional configuration for Elasticsearch yaml configuration file. Each line must be separated by a newline character <code>\n</code> e.g. <code>"action.auto_create_index: .security\nindices.queries.cache.size: 5%"</code>. <strong>This is an expert level feature - It is recommended that you run your additional yaml through a <a href="http://www.yamllint.com/">linter</a> before starting a deployment.</strong>
     </td><td><code>""</code></td></tr>
 
+  <tr><td>esHeapSize</td><td>integer</td>
+    <td>The size, in megaytes, of memory to allocate on each Elasticsearch node for the JVM heap. If unspecified, 50% of the available memory will be allocated to Elasticsearch heap, up to a maximum of 31744 MB.
```
Comment (codebrain) [inline README.md]: megaytes.... erm :)

Comment (codebrain) [reply README.md]: up to a maximum of 31744 MB (32GB)

File: src/mainTemplate.json [hunk 2]
```diff
@@ -116,6 +118,13 @@
         "description": "Additional configuration for Elasticsearch yaml configuration file. Each line must be separated by a newline character e.g. action.auto_create_index: .security\nindices.queries.cache.size:5%"
       }
     },
+    "esHeapSize": {
+      "type": "int",
+      "defaultValue": 0,
+      "metadata": {
+        "description": "The size, in megabytes, of memory to allocate on each Elasticsearch node for the JVM heap. If unspecified, 50% of the available memory will be allocated to Elasticsearch heap, up to a maximum of 31744 MB."
```
Comment (codebrain) [inline src/mainTemplate.json]: up to a maximum of 31744 MB (32GB)

**Pull request merged**